packages/react-checkbox/src/components/Checkbox/useCheckbox.tsx

```diff
@@ -40,7 +40,7 @@ export const useCheckbox_unstable = (props: CheckboxProps, ref: React.Ref<HTMLIn
   const nativeProps = getPartitionedNativeProps({
     props,
     primarySlotTagName: 'input',
-    excludedPropNames: ['checked', 'defaultChecked', 'size'],
+    excludedPropNames: ['checked', 'defaultChecked', 'size', 'onChange'],
```

Comment:
_Curiosity_: why this change is required?

Reply:
Added to the PR description:
> This PR improved the type safety if `resolveShorthand`, and it caught a few mistakes in the use of `getPartitionedNativeProps` when a component has a prop whose type differs from the native prop (e.g. Checkbox adds a param to the `onChange` function). I added a few excludedPropNames to components that use `getPartitionedNativeProps` to fix this.